GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -412,19 +412,23 @@ protected override void OnActivated(EventArgs e)
             base.OnActivated(e);
         }
 
-        protected override void OnFormClosing(FormClosingEventArgs e)
+        protected override void OnFormClosed(FormClosedEventArgs e)
         {
-            // Do not remember commit message of fixup or squash commits, since they have
-            // a special meaning, and can be dangerous if used inappropriately.
-            if (CommitKind == CommitKind.Normal)
+            // Do not attempt to store again if the form has already been closed. Unfortunately, OnFormClosed is always called by Close.
+            if (Visible)
```

Comment:
Are you seeing a re-entrancy here?

Reply:
`Close` can be called multiple times by the unit tests in order to ensure the forms are disposed before the next test case.

Reply:
Yes, then `OnFormClosing` and `OnFormClose` are called everytime.